### PR TITLE
Add ci-notify to vmimage build.

### PR DESCRIPTION
**Pull request by kwoodson**
Adding ci notifications to vmbuild.

```release-note:
None
```

### CODE REVIEW

File: hack/vmimage.sh [hunk 1]
```diff
@@ -18,25 +25,32 @@ else
   GITCOMMIT="$TAG-dirty"
 fi
 
+false
```
Comment (mjudeikis) [inline hack/vmimage.sh]: what is this?

File: hack/ci-notify/main.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		fmt.Print("No issue found.  Creating one")
+		ir := &github.IssueRequest{
+			Title:     to.StringPtr(jobName),
+			Body:      to.StringPtr(fmt.Sprintf("The %s build has failed.  Please check the following [link](%s)<br/><br/>%s", jobName, getJobLogsURL(jobName), comment)),
+			Assignees: to.StringSlicePtr(users),
+		}
+		_, _, err = c.gh.Issues.Create(ctx, owner, repo, ir)
+		if err != nil {
+			return err
+		}
```
Comment (mjudeikis) [inline hack/ci-notify/main.go]: return `nil` and drop `else` ?

File: hack/vmimage.sh [hunk 3]
```diff
@@ -3,12 +3,19 @@
 cleanup() {
   set +e
 
+  if [[ -e /var/run/secrets/kubernetes.io ]]; then
+    if [ $PHASE == "build_complete" ]; then
+      ARGS="-success"
+    fi
+    go run ./hack/ci-notify/main.go -job-name periodic-ci-azure-vmimage -comment "Phase: $PHASE" $ARGS
```
Comment (m1kola) [inline hack/vmimage.sh]: Do we need to hard code `periodic-ci-azure-vmimage` here? Can we replace it with `$JOB_NAME`?

File: hack/vmimage.sh [hunk 4]
```diff
@@ -3,12 +3,19 @@
 cleanup() {
   set +e
 
+  if [[ -e /var/run/secrets/kubernetes.io ]]; then
```
Comment (m1kola) [inline hack/vmimage.sh]: Notification is not part of a cleanup. Can we move this check into a separate function-trap?

File: hack/ci-notify/main.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+	issues, _, err := c.gh.Issues.ListByRepo(ctx, owner, repo, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	return issues, nil
+}
+
+func (c *client) getVMImageBuildIssue(ctx context.Context, jobName, owner, repo string, since time.Duration) (*github.Issue, error) {
```
Comment (m1kola) [inline hack/ci-notify/main.go]: The name is very specific. I think we will be using this script not only for `vmimage` builds (`vmimage-validate`, for example). Maybe`getIssueByName` or something like that?

File: hack/vmimage.sh [hunk 6]
```diff
@@ -3,12 +3,20 @@
 cleanup() {
   set +e
 
+  if [[ -e /var/run/secrets/kubernetes.io ]] && [ "${JOB_TYPE} == "periodic" ]; then
+    if [ $PHASE == "build_complete" ]; then
```
Comment (m1kola) [inline hack/vmimage.sh]: In traps we can check exit code with with `$?` ;)

Comment (m1kola) [reply hack/vmimage.sh]: Ah, right! We can make it work somehow like this:

```bash
    exit_code=$?
    set +e
```

But I'm not sure if the efforts worth it.

**Pull request merged**